### PR TITLE
new recipe for ConnectionPool

### PR DESCRIPTION
**Pull request by jmeppley**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
The original developer hasn't responded to past pull requests, so I forked the project. This is a small little thing, but I think it could be useful to keep available as a separate project rather than just folding the code into where I'm going to use it.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
  (N/A)
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Attn: @conda-forge/help-python

### CODE REVIEW

File: recipes/connectionpool/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+requirements:
+  build:
+    - python
+
+  run:
+    - python
+
+about:
+  home: https://github.com/{{ author }}/{{ name }}
+  # Remember to specify the license variants for BSD, Apache, GPL, and LGPL.
```
Comment (scopatz) [inline recipes/connectionpool/meta.yaml:35]: Please remove comments

File: recipes/connectionpool/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  home: https://github.com/jmeppley/{{ name }}
+  summary: {{ summary }}
+
+build:
+  number: 0
+  noarch: python
+  script: $PYTHON -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build:
```
Comment (scopatz) [inline recipes/connectionpool/meta.yaml:27]: This should be host

File: recipes/connectionpool/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    - python
```
Comment (scopatz) [inline recipes/connectionpool/meta.yaml:28]: you probably need pip here

File: recipes/connectionpool/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+source:
+    fn: {{ name }}.{{ version }}.tar.gz
+    url: https://github.com/jmeppley/{{ name }}/archive/v{{ version }}.tar.gz
+    sha256: {{ sha256 }}
+
+
+about:
+  home: https://github.com/jmeppley/{{ name }}
+  summary: {{ summary }}
```
Comment (scopatz) [inline recipes/connectionpool/meta.yaml:19]: You probably don't need two about sections

File: recipes/connectionpool/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{% set name = "ConnectionPool" %}
```
Comment (synapticarbors) [inline recipes/connectionpool/meta.yaml:1]: If you follow the example meta.yaml (https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml), this should look something like:

```
{% set name = "ConnectionPool" %}
{% set version = "0.1.0" %}

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
    url: https://github.com/jmeppley/{{ name }}/archive/v{{ version }}.tar.gz
    sha256: 123861f0ff0b7d0967d2fb3cbefd13a461091be7a5451942f820c7f17062669f"

build:
  number: 0
  noarch: python
  script: "{{ PYTHON }} -m pip install . -vv"

requirements:
  host:
    - python
    - pip
  run:
    - python

about:
  home: https://github.com/jmeppley/ConnectionPool
  license: MIT
  license_family: MIT
  license_file: LICENSE
  summary: 'A simple thread-safe pure-python connection pool'

extra:
  recipe-maintainers:
    - jmeppley
```

**Review by synapticarbors — COMMENTED** [review]
I ended up just giving you what I think is a complete sample of what the recipe should look like. Also greetings from a fellow Haverford grad!

**Comment by synapticarbors** [comment]
Also since the original package of the same name is up on pypi.io (https://pypi.org/project/connection-pool/) and you have a fork, I think we might want to be careful with the naming of the package so that there isn't confusion, unless the original project you forked from is known to no longer be developed and this is the official new branch of development.

File: recipes/connectionpool/meta.yaml (hunk 5 above)
Comment (jmeppley) [reply recipes/connectionpool/meta.yaml:1]: From the looks of it, we overlapped for a year at HC. I was '98.

**Comment by jmeppley** [comment]
Thanks for the feedback, @synapticarbors  and @scopatz.

I just noticed that I changed the package name slightly from the original pip. I meant to use the same name.

I'm not sure of the etiquette of taking on this project. I don't think it needs to be forked, but It seems like the original developer might have abandoned it, although it's only been a couple years. I've reached out to them and can wait a bit to see if I get a response.

Do either of you have any suggestions on how to proceed?

Either way, I should probably close this and re-open with either the pip name or a sufficiently different name.

**Pull request closed without merging**